### PR TITLE
[5.3] Micro optimizations in query builder

### PR DESCRIPTION
**Pull request by vlakoff**
`empty()` is a better pick than `count()`.

Refs #16188 (merged to master, will conflict)

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -768,7 +768,7 @@ public function forNestedWhere()
      */
     public function addNestedWhereQuery($query, $boolean = 'and')
     {
-        if (count($query->wheres)) {
```
Comment (vlakoff) [inline src/Illuminate/Database/Query/Builder.php]: Note here we were doing `count(null)` (gives 0). Not ideal…

**Review by vlakoff — COMMENTED** [review]
Use query string `?w=0` to better view this diff.

File: src/Illuminate/Database/Connection.php [hunk 2]
```diff
@@ -290,13 +290,13 @@ public function raw($value)
      *
      * @param  string  $query
      * @param  array   $bindings
-     * @return mixed
+     * @return mixed|null
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Connection.php]: null is a subset of mixed, don't add it

File: src/Illuminate/Database/Query/Builder.php [hunk 3]
```diff
@@ -1597,13 +1597,13 @@ public function find($id, $columns = ['*'])
      * Get a single column's value from the first result of a query.
      *
      * @param  string  $column
-     * @return mixed
+     * @return mixed|null
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: same

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Database/Connection.php (hunk 2 above)
Comment (vlakoff) [inline src/Illuminate/Database/Connection.php]: Was expecting this, lol. PR updated.

**Comment by taylorotwell** [comment]
Honestly nothing is broken here and I have a feeling these are very micro optimizations. I don't want to get obsessed about this kind of stuff throughout the framework.

**Pull request closed without merging**